Makefile

```diff
@@ -180,6 +181,15 @@ learn: install | $(NODELIB)
 	levy export html --no-self-contained --root-dir="/learn/" learn/content $(LEARN_TARGET_DIR)
 	npx --prefix $(NODEPREFIX) -y prettier --write $(LEARN_TARGET_DIR)
 
+## Generate SVG files from .evy files for lab.evy.dev in frontend/lab
+lab: | $(NODELIB)
+	@for n in $$(fd --full-path --glob '**/img/*.evy' $(LAB_TARGET_DIR)); do \
+	  echo "generating $${n%.evy}.svg"; \
+	  evy run --svg-out "$${n%.evy}.svg" "$$n"; \
+	done
```

Comment:
Another way to do this:

```
lab: IMAGES := $(shell fd --full-path --glob '**/img/*.evy' $(LAB_TARGET_DIR))
lab: | $(NODELIB)
        $(foreach image,$(IMAGES),evy run --svg-out "$(image:evy=svg)" "$(image)"$(nl))
        npx --prefix $(NODEPREFIX) -y prettier --write $(LAB_TARGET_DIR)
```

It may just be a matter of taste, but I tend to prefer doing the loop in Make if it is not too ugly because the result is that the make recipe is just a linear list of commands to run, and when one fails, that is where make will stop. With the loop in sh, you don't always know where in the loop it is. It also works a bit better with `make -n`:
```
make -n lab
evy run --svg-out "frontend/lab/samples/intro/img/circles-anika.svg" "frontend/lab/samples/intro/img/circles-anika.evy"
evy run --svg-out "frontend/lab/samples/intro/img/circles-bottom-left.svg" "frontend/lab/samples/intro/img/circles-bottom-left.evy"
evy run --svg-out "frontend/lab/samples/intro/img/circles-bottom-right.svg" "frontend/lab/samples/intro/img/circles-bottom-right.evy"
evy run --svg-out "frontend/lab/samples/intro/img/circles-camh.svg" "frontend/lab/samples/intro/img/circles-camh.evy"
evy run --svg-out "frontend/lab/samples/intro/img/circles-kathi.svg" "frontend/lab/samples/intro/img/circles-kathi.evy"
evy run --svg-out "frontend/lab/samples/intro/img/circles-mali.svg" "frontend/lab/samples/intro/img/circles-mali.evy"
evy run --svg-out "frontend/lab/samples/intro/img/circles-top-left.svg" "frontend/lab/samples/intro/img/circles-top-left.evy"
evy run --svg-out "frontend/lab/samples/intro/img/circles-top-right.svg" "frontend/lab/samples/intro/img/circles-top-right.evy"
evy run --svg-out "frontend/lab/samples/intro/img/coords-30-60.svg" "frontend/lab/samples/intro/img/coords-30-60.evy"
evy run --svg-out "frontend/lab/samples/intro/img/move-6circles.svg" "frontend/lab/samples/intro/img/move-6circles.evy"
npx --prefix .hermit/node -y prettier --write frontend/lab
```

You can't do it when the first part of the recipe creates the files that you then need to find, but in this case that's not a problem.

Also, because the shell is not being invoked with `-e`, in this recipe in this PR, if there is an error with one of the files, it will continue to the rest and then probably sucessfully complete.

Anyway, food for thought. Up to you :pineapple:

Reply:
ok, looks better. thank you, changed.